### PR TITLE
Updated test code using test data matrix generation of test-unit 3.2.9.

### PR DESCRIPTION
**Pull request by naitoh**
Updated test code using test data matrix generation of test-unit 3.2.9.

Please refer to the following for improvement contents of test-unit 3.2.9.
* https://github.com/test-unit/test-unit/blob/master/doc/text/news.md

### CODE REVIEW

File: test/functions/activation/leaky_relu_test.rb [hunk 1]
```diff
@@ -3,13 +3,11 @@
...
-    'test1' => {shape: [3, 2], dtype: xm::SFloat},
-    'test2' => {shape: [], dtype: xm::SFloat},
-    'test3' => {shape: [3, 2], dtype: xm::DFloat},
-    'test4' => {shape: [], dtype: xm::DFloat}}
 
-  def _setup(data)
+  data(:shape, [[3, 2], []],             keep: true)
+  data(:dtype, [xm::SFloat, xm::DFloat], keep: true)
+
+  def setup()
```
Comment (sonots) [inline test/functions/activation/leaky_relu_test.rb]: Do you need ()?
It looks not rubyish.

Comment (sonots) [reply test/functions/activation/leaky_relu_test.rb]: ditto for others

Comment (naitoh) [reply test/functions/activation/leaky_relu_test.rb]: > Do you need ()?

Does this point to **setup()** -> **setup**?
Or does this point to **data(:shape, [[3, 2], []],             keep: true)** -> **data :shape, [[3, 2], []],             keep: true**?

Comment (sonots) [reply test/functions/activation/leaky_relu_test.rb]: Of course, 1. All def lines

File: test/functions/loss/softmax_cross_entropy_test.rb [hunk 2]
```diff
@@ -237,23 +212,17 @@ def check_backward(x_data, t_data, g_data, class_weight)
...
   end
 end
 
 class TestSoftmaxCrossEntropyInvalidReduce < Test::Unit::TestCase
-  use_cudnn    = ['always', 'auto', 'never'],
-  normalize    = [true, false]
-  cache_score  = [true, false]
+  data(:use_cudnn,   ['always', 'auto', 'never'], group: :reduce, keep: true)
+  data(:normalize,   [true, false],               group: :reduce, keep: true)
+  data(:cache_score, [true, false],               group: :reduce, keep: true)
```
Comment (kou) [inline test/functions/loss/softmax_cross_entropy_test.rb]: We can drop `group:` when we only use one data matrix group.

File: test/functions/loss/softmax_cross_entropy_test.rb [hunk 3]
```diff
@@ -268,22 +237,16 @@ def check_invalid_reduce(x, t)
...
 
 class TestNonDefaultIgnoreLabel < Test::Unit::TestCase
-  reduce       = ['mean', 'no']
-  class_weight = [nil, xm::SFloat.ones([3])]
-
-  data = reduce.product(class_weight).collect {|v|
-    {reduce: v[0], class_weight: v[1]}}
-  data = (1..data.size).to_a.zip(data).to_h
+  data(:reduce,       ['mean', 'no'],              group: :reduce, keep: true)
+  data(:class_weight, [nil, xm::SFloat.ones([3])], group: :reduce, keep: true)
```
Comment (kou) [inline test/functions/loss/softmax_cross_entropy_test.rb]: ditto.

File: test/functions/loss/softmax_cross_entropy_test.rb (hunk 2 above)
Comment (naitoh) [reply test/functions/loss/softmax_cross_entropy_test.rb]: Thank you for pointing out.
I fixed it.

File: test/functions/loss/softmax_cross_entropy_test.rb (hunk 3 above)
Comment (naitoh) [reply test/functions/loss/softmax_cross_entropy_test.rb]: Thank you for pointing out.
I fixed it.

**Comment by naitoh** [comment]
I fixed the pointed out items, so please confirm.

**Comment by naitoh** [comment]
@kou Is there anything to point out about this fix?

**Comment by naitoh** [comment]
Thanks!

**Pull request merged**